python_modules/dagster/dagster/_core/execution/plan/execute_step.py

```diff
@@ -705,51 +705,56 @@ def _store_output(
     manager_materializations = []
     manager_metadata: Dict[str, MetadataValue] = {}
 
-    # output_manager.handle_output is either a generator function, or a normal function with or
-    # without a return value. In the case that handle_output is a normal function, we need to
-    # catch errors should they be raised before a return value. We can do this by wrapping
-    # handle_output in a generator so that errors will be caught within iterate_with_context.
-
-    if not inspect.isgeneratorfunction(output_manager.handle_output):
-
-        def _gen_fn():
-            gen_output = output_manager.handle_output(output_context, output.value)
+    # don't store asset check outputs
+    if not step_context.step.step_output_named(
+        step_output_handle.output_name
+    ).properties.asset_check_handle:
```

Comment:
nit: not a fan of this large "common path" block nested in "if not special case"

you could assign `handle_output_gen` to a no op fn, like done here https://github.com/dagster-io/dagster/pull/15786/files